Mathlib/Analysis/Fourier/FourierTransformDeriv.lean

```diff
@@ -152,17 +167,171 @@ theorem hasFDerivAt_fourierIntegral
     (hf : Integrable f μ) (hf' : Integrable (fun v : V ↦ ‖v‖ * ‖f v‖) μ) :
     Differentiable ℝ (fourierIntegral 𝐞 μ L.toLinearMap₂ f) :=
   fun w ↦ (hasFDerivAt_fourierIntegral L hf hf' w).differentiableAt
 
+/-- The formal multilinear series whose `n`-th term is
+`(w₁, ..., wₙ) ↦ (-2πI)^n * L v w₁ * ... * L v wₙ • f v`, as a continuous multilinear map in
+the space `W [×n]→L[ℝ] E`.
+
+This is designed so that the Fourier transform of `v ↦ fourierPowSMulRight L f v n` is the
+`n`-th derivative of the Fourier transform of `f`.
+-/
+def fourierPowSMulRight (f : V → E) (v : V) : FormalMultilinearSeries ℝ W E := fun n ↦
+  (- (2 * π * I))^n • ((ContinuousMultilinearMap.mkPiRing ℝ (Fin n) (f v)).compContinuousLinearMap
+  (fun _i ↦ L v))
+
+@[simp] lemma fourierPowSMulRight_apply {f : V → E} {v : V} {n : ℕ} {m : Fin n → W} :
+    fourierPowSMulRight L f v n m = (- (2 * π * I))^n • (∏ i, L v (m i)) • f v := by
+  simp [fourierPowSMulRight]
+
+open ContinuousMultilinearMap
+
+/-- Decomposing `fourierPowSMulRight L f v n` as a composition of continuous bilinear and
+multilinear maps, to deduce easily its continuity and differentiability properties. -/
+lemma fourierPowSMulRight_eq_comp {f : V → E} {v : V} {n : ℕ} :
+    fourierPowSMulRight L f v n = (- (2 * π * I))^n • smulRightL ℝ (fun (_ : Fin n) ↦ W) E
+      (compContinuousLinearMapLRight
+        (ContinuousMultilinearMap.mkPiAlgebra ℝ (Fin n) ℝ) (fun _ ↦ L v)) (f v) := rfl
+
+@[continuity, fun_prop]
+lemma _root_.Continuous.fourierPowSMulRight {f : V → E} (hf : Continuous f) (n : ℕ) :
+    Continuous (fun v ↦ fourierPowSMulRight L f v n) := by
+  simp_rw [fourierPowSMulRight_eq_comp]
+  apply Continuous.const_smul
+  apply (smulRightL ℝ (fun (_ : Fin n) ↦ W) E).continuous₂.comp₂ _ hf
+  exact Continuous.comp (map_continuous _) (continuous_pi (fun _i ↦ L.continuous))
+
+lemma _root_.ContDiff.fourierPowSMulRight {f : V → E} {k : ℕ∞} (hf : ContDiff ℝ k f) (n : ℕ) :
+    ContDiff ℝ k (fun v ↦ fourierPowSMulRight L f v n) := by
+  simp_rw [fourierPowSMulRight_eq_comp]
+  apply ContDiff.const_smul
+  apply (smulRightL ℝ (fun (_ : Fin n) ↦ W) E).isBoundedBilinearMap.contDiff.comp₂ _ hf
+  apply (ContinuousMultilinearMap.contDiff _).comp
+  exact contDiff_pi.2 (fun _ ↦ L.contDiff)
+
+lemma norm_fourierPowSMulRight_le (f : V → E) (v : V) (n : ℕ) :
+    ‖fourierPowSMulRight L f v n‖ ≤ (2 * π * ‖L‖) ^ n * ‖v‖ ^ n * ‖f v‖ := by
+  apply ContinuousMultilinearMap.opNorm_le_bound _ (by positivity) (fun m ↦ ?_)
+  calc
+  ‖fourierPowSMulRight L f v n m‖
+    = (2 * π) ^ n * ((∏ x : Fin n, |(L v) (m x)|) * ‖f v‖) := by
+      simp [_root_.abs_of_nonneg pi_nonneg, norm_smul]
+  _ ≤ (2 * π) ^ n * ((∏ x : Fin n, ‖L‖ * ‖v‖ * ‖m x‖) * ‖f v‖) := by
+      gcongr with i _hi
+      · exact fun i _hi ↦ abs_nonneg _
+      · exact L.le_opNorm₂ v (m i)
+  _ = (2 * π * ‖L‖) ^ n * ‖v‖ ^ n * ‖f v‖ * ∏ i : Fin n, ‖m i‖ := by
+      simp [Finset.prod_mul_distrib, mul_pow]; ring
+
+variable {L}
```

Comment:
why did you make L implicit for the next two lemmas ? I mean why for those two specifically, since all others have explicit L.

Reply:
No good reason. I've made it explicit.